eessi/testsuite/tests/apps/osu.py

```diff
@@ -0,0 +1,314 @@
+            # This if condition had to be added since the CUDA compiled osu
+            # tests do not run on cpu partitions. The binaries need
+            # libcuda.so.1 during runtime which can only be found in a
+            # partition with CUDA drivers.
+            self.valid_systems = [f'+{FEATURES[CPU]} +{FEATURES[GPU]} %{GPU_VENDOR}={GPU_VENDORS[NVIDIA]}']
+
+        # If the device_type is CPU then device buffer should always be CPU.
+        if self.device_type == DEVICE_TYPES[CPU]:
+            self.device_buffers = 'cpu'
+
+        # This part of the code removes the collective communication calls out
+        # of the run list since this test is only meant for pt2pt.
+        if not self.benchmark_info[0].startswith('mpi.pt2pt'):
+            self.valid_systems = []
+        hooks.set_modules(self)
+
+    @run_after('init')
+    def set_tag_ci(self):
+        """ Setting tests under CI tag. """
+        if (self.benchmark_info[0] in ['mpi.pt2pt.osu_latency',
+                                       'mpi.pt2pt.osu_bw']):
+            self.tags.add('CI')
+            log(f'tags set to {self.tags}')
+
+        if (self.benchmark_info[0] == 'mpi.pt2pt.osu_bw'):
+            self.tags.add('osu_bw')
+
+        if (self.benchmark_info[0] == 'mpi.pt2pt.osu_latency'):
+            self.tags.add('osu_latency')
+
+    @run_after('init')
+    def set_mem(self):
+        """ Setting an extra job option of memory. This test has only 4
+        possibilities: 1_node, 2_nodes, 2_cores and 1_cpn_2_nodes. Only the
+        last 2 require the memory to be set. """
+        if(SCALES.get(self.scale).get('node_part', 0) == 0):
+            self.extra_resources = {'memory': {'size': '32GB'}}
+
+    @run_after('init')
+    def set_num_tasks(self):
+        """ Setting scales as tags. """
+        hooks.set_tag_scale(self)
+
+    @run_after('setup')
+    def set_num_tasks_per_node(self):
+        """ Setting number of tasks per node and cpus per task in this function.
+        This function sets num_cpus_per_task for 1 node and 2 node options where
+        the request is for full nodes."""
+        if(SCALES.get(self.scale).get('num_nodes') == 1):
+            hooks.assign_tasks_per_compute_unit(self,
+                                                   COMPUTE_UNIT.get(NODE,
+                                                                    'node'), 2)
+        else:
+            hooks.assign_tasks_per_compute_unit(self, COMPUTE_UNIT.get(NODE,
+                                                                       'node'))
+
+    @run_after('setup')
+    def set_num_gpus_per_node(self):
+        """
+        This test does not require gpus and is for host to host within GPU
+        nodes. But some systems do require a GPU allocation for to perform any
+        activity in the GPU nodes.
+        """
+        if('gpu' in self.current_partition.features and
+           not utils.is_cuda_required_module(self.module_name)):
```

Comment:
On Snellius, one needs to allocate at least 1 GPU if a node is to be granted in a `gpu` partition. This can be a policy at other clusters as well. Therefore, I have allocated 1 GPU even for non-cuda modules.